rasterio/_features.pyx

```diff
@@ -357,12 +363,15 @@ def _rasterize(shapes, image, transform, all_touched, merge_alg):
                     NULL, pixel_values, options, NULL, NULL))
 
     finally:
-        for i in range(<int>num_geoms):
-            _deleteOgrGeom(geoms[i])
-        CPLFree(geoms)
-        CPLFree(pixel_values)
-        CPLFree(band_ids)
-        if options:
+        if geoms != NULL:
+            for i in range(<int>num_geoms):
+                _deleteOgrGeom(geoms[i])
+            CPLFree(geoms)
+        if pixel_values != NULL:
+            CPLFree(pixel_values)
```

Comment:
@snowman2 CPLFree (which is an alias of VSIFree) is a no-op if the arg is NULL. See https://github.com/OSGeo/gdal/blob/master/port/cpl_vsisimple.cpp#L872. So the guard isn't need.

Reply:
Thanks 👍. I updated it.